src/transforms/transform-pull-request.ts

```diff
@@ -52,8 +50,8 @@ const mapReviews = async (
 
 			acc.push(usernames[reviewerUsername]);
 
-		} else if (usernames[reviewerUsername].approvalStatus !== STATE_APPROVED && review.state === STATE_APPROVED) {
-			usernames[reviewerUsername].approvalStatus = STATE_APPROVED;
+		} else {
+			usernames[reviewerUsername].approvalStatus = review.state === STATE_APPROVED ? STATE_APPROVED : STATE_UNAPPROVED;
```

Comment:
If we've already seen a reviewer, all we want to do is make sure approval status is updated to their most recent review.state

Reply:
Before we were basically checking if we had already seen a reviewer before or not. If not, we created a usernames object. Otherwise, if we had seen a reviewer before, we only updated the review state if the approvalStatus was UNAPPROVED and had since changed to APPROVED. If it was the opposite e.g. first reviewer APPROVED (state stored to approvalStatus) but then requested changes (UNAPPROVED) we did... nothing. Which means the approvalStatus remained unchanged.

Essentially, what we want to do is:

- check if we've seen the reviewer before: if not, create the usernames object and store APPROVED/UNAPPROVED in approvalStatus
- otherwise, update approvalStatus to the most recent review.state